src/ProblemDetails/IHttpProblemDetails.cs

```diff
@@ -0,0 +1,46 @@
+﻿// <copyright file="IHttpProblemDetails.cs" company="Cognisant Research">
+// Copyright (c) Cognisant Research. All rights reserved.
+// </copyright>
+
+namespace CR.ProblemDetails
+{
+    using System;
+    using System.Net;
+
+    /// <summary>
+    /// A description of a problem which occurred while fulfilling a HTTP request.
+    /// </summary>
+    public interface IHttpProblemDetails
```

Comment:
The spec supports extensions, so extra properties on the problem details, we should support this too.